### PR TITLE
test: migrated use-semicolon-delimiter.test.js from tap to node:test

### PR DESCRIPTION
**Pull request by Tony133**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Proposal:

  - rename file in `kebab-case`
  - migrated `use-semicolon-delimiter.js` file from `tap` to `node:test` 🔥

### CODE REVIEW

**Review by Uzlopak — APPROVED** [review]

**Comment by Uzlopak** [comment]
please dont force push. it is not possible to see what significant changes you did. at the end we will anyway squash merge and have it all in once commit

**Comment by Tony133** [comment]
> please dont force push. it is not possible to see what significant changes you did. at the end we will anyway squash merge and have it all in once commit

ok 👍

File: test/use-semicolon-delimiter.test.js [hunk 1]
```diff
@@ -1,110 +1,101 @@
 'use strict'
 
-const t = require('tap')
-const test = t.test
+const { test } = require('node:test')
 const Fastify = require('..')
 const sget = require('simple-get').concat
 
 test('use semicolon delimiter default false', t => {
```
Comment (dancastillo) [inline test/use-semicolon-delimiter.test.js:7]: Add in `done` so we assert the number of tests planned for callbacks
```js
test('use semicolon delimiter default false', (t, done) => {
  t.plan(4)
  const fastify = Fastify({})

  t.after(fastify.close())

  fastify.get('/1234;foo=bar', (req, reply) => {
    reply.send(req.query)
  })

  fastify.listen({ port: 0 }, err => {
    t.assert.ifError(err)

    sget({
      method: 'GET',
      url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
    }, (err, response, body) => {
      t.assert.ifError(err)
      t.assert.strictEqual(response.statusCode, 200)
      t.assert.deepStrictEqual(JSON.parse(body), {})
      done()
    })
  })
})
```

File: test/use-semicolon-delimiter.test.js [hunk 2]
```diff
@@ -1,110 +1,101 @@
...
-      t.equal(response.statusCode, 200)
-      t.same(JSON.parse(body), {})
+      t.assert.ifError(err)
+      t.assert.strictEqual(response.statusCode, 200)
+      t.assert.deepStrictEqual(JSON.parse(body), {})
     })
   })
 })
 
 test('use semicolon delimiter set to true', t => {
```
Comment (dancastillo) [inline test/use-semicolon-delimiter.test.js:30]: Call `done` in after last test assertion
```suggestion
test('use semicolon delimiter set to true', (t, done) => {
  t.plan(4)
```

File: test/use-semicolon-delimiter.test.js [hunk 3]
```diff
@@ -1,110 +1,101 @@
...
+      t.assert.ifError(err)
+      t.assert.strictEqual(response.statusCode, 200)
+      t.assert.deepStrictEqual(JSON.parse(body), {
         foo: 'bar'
       })
     })
   })
 })
 
 test('use semicolon delimiter set to false', t => {
```
Comment (dancastillo) [inline test/use-semicolon-delimiter.test.js:56]: Call `done` in after last test assertion
```suggestion
test('use semicolon delimiter set to false', (t, done) => {
  t.plan(4)
```

File: test/use-semicolon-delimiter.test.js [hunk 4]
```diff
@@ -1,110 +1,101 @@
...
-      t.equal(response.statusCode, 200)
-      t.same(JSON.parse(body), {})
+      t.assert.ifError(err)
+      t.assert.strictEqual(response.statusCode, 200)
+      t.assert.deepStrictEqual(JSON.parse(body), {})
     })
   })
 })
 
 test('use semicolon delimiter set to false return 404', t => {
```
Comment (dancastillo) [inline test/use-semicolon-delimiter.test.js:80]: Call `done` in after last test assertion
```suggestion
test('use semicolon delimiter set to false return 404', (t, done) => {
  t.plan(3)
```

Comment (Tony133) [reply test/use-semicolon-delimiter.test.js]: Adding the `done()` returns me this:

```
Error [ERR_TEST_FAILURE]: The resolution of the promise is still pending but the event loop has already been resolved
at process.emit (node:events:519:28) {
code: ‘ERR_TEST_FAILURE’,
failureType: ‘cancelledByParent’,
cause: ‘The resolution of the promise is still in progress, but the event loop has already been resolved’.
}
 ```

Comment (dancastillo) [reply test/use-semicolon-delimiter.test.js]: Interesting not sure why that is not working....

This works calling `fastify.close()` before `done()`
```js
'use strict'

const { test } = require('node:test')
const Fastify = require('..')
const sget = require('simple-get').concat

test('use semicolon delimiter default false', (t, done) => {
  t.plan(4)

  const fastify = Fastify()

  fastify.get('/1234;foo=bar', (req, reply) => {
    console.log('helloooo')
    reply.send(req.query)
  })

  fastify.listen({ port: 0,  }, err => {
    t.assert.ifError(err)
    sget({
      method: 'GET',
      url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
    }, (err, response, body) => {
        console.error(err)
      t.assert.ifError(err)
      t.assert.strictEqual(response.statusCode, 200)
      t.assert.deepStrictEqual(JSON.parse(body), {})
      fastify.close()
      done()
    })
  })
})

test('use semicolon delimiter set to true', (t, done) => {
  t.plan(4)
  const fastify = Fastify({
    useSemicolonDelimiter: true
  })

  fastify.get('/1234', (req, reply) => {
    reply.send(req.query)
  })

  fastify.listen({ port: 0 }, err => {
    t.assert.ifError(err)

    sget({
      method: 'GET',
      url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
    }, (err, response, body) => {
      t.assert.ifError(err)
      t.assert.strictEqual(response.statusCode, 200)
      t.assert.deepStrictEqual(JSON.parse(body), {
        foo: 'bar'
      })
      fastify.close()
      done()
    })
  })
})

test('use semicolon delimiter set to false', (t, done) => {
  t.plan(4)

  const fastify = Fastify({
    useSemicolonDelimiter: false
  })

  fastify.get('/1234;foo=bar', (req, reply) => {
    reply.send(req.query)
  })

  fastify.listen({ port: 0 }, err => {
    t.assert.ifError(err)

    sget({
      method: 'GET',
      url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
    }, (err, response, body) => {
      t.assert.ifError(err)
      t.assert.strictEqual(response.statusCode, 200)
      t.assert.deepStrictEqual(JSON.parse(body), {})
      fastify.close()
      done()
    })
  })
})

test('use semicolon delimiter set to false return 404', (t, done) => {
  t.plan(3)

  const fastify = Fastify({
    useSemicolonDelimiter: false
  })

  fastify.get('/1234', (req, reply) => {
    reply.send(req.query)
  })

  fastify.listen({ port: 0 }, err => {
    t.assert.ifError(err)

    sget({
      method: 'GET',
      url: 'http://localhost:' + fastify.server.address().port + '/1234;foo=bar'
    }, (err, response, body) => {
      t.assert.ifError(err)
      t.assert.strictEqual(response.statusCode, 404)
      fastify.close()
      done()
    })
  })
})
```

Comment (Tony133) [reply test/use-semicolon-delimiter.test.js]: very interesting

Comment (Tony133) [reply test/use-semicolon-delimiter.test.js]: ok, now the test passes all the checks, we will see if this solution will be the final one.

Thanks for the help 🙂

**Review by Eomm — APPROVED** [review]

**Comment by jsumners** [comment]
> please dont force push. it is not possible to see what significant changes you did. at the end we will anyway squash merge and have it all in once commit

GitHub provides a compare tool to see what changes have been made during a force push.

**Pull request merged**